fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsJob.java

```diff
@@ -18,62 +18,187 @@
+    private final long createTime = System.currentTimeMillis();
+    private long scheduleTime = -1L;
+    private long finishTime = -1L;
+    private int progress = 0;
+
+    public StatisticsJob(Long dbId,
+                         List<Long> tableIdList,
+                         Map<Long, List<String>> tableIdToColumnName,
+                         Map<String, String> properties) {
+        this.dbId = dbId;
+        this.tableIds = tableIdList;
+        this.tableIdToColumnName = tableIdToColumnName;
+        this.properties = properties;
+    }
 
     public long getId() {
-        return id;
+        return this.id;
+    }
+
+    public long getDbId() {
+        return this.dbId;
+    }
+
+    public List<Long> getTableIds() {
+        return this.tableIds;
+    }
+
+    public Map<Long, List<String>> getTableIdToColumnName() {
+        return this.tableIdToColumnName;
+    }
+
+    public Map<String, String> getProperties() {
+        return this.properties;
+    }
+
+    public List<StatisticsTask> getTasks() {
+        return this.tasks;
+    }
+
+    public void setTasks(List<StatisticsTask> tasks) {
+        this.tasks = tasks;
+    }
+
+    public JobState getJobState() {
+        return this.jobState;
+    }
+
+    public void setJobState(JobState jobState) {
+        this.jobState = jobState;
+    }
+
+    public long getCreateTime() {
+        return this.createTime;
+    }
+
+    public long getScheduleTime() {
+        return this.scheduleTime;
+    }
+
+    public void setScheduleTime(long scheduleTime) {
+        this.scheduleTime = scheduleTime;
     }
 
-    /*
-        AnalyzeStmt: Analyze t1(c1), t2
-        StatisticsJob:
-          tableId [t1, t2]
-          tableIdToColumnName <t1, [c1]> <t2, [c1,c2,c3]>
-         */
-    public static StatisticsJob fromAnalyzeStmt(AnalyzeStmt analyzeStmt) {
-        // TODO
-        return new StatisticsJob();
+    public long getFinishTime() {
+        return this.finishTime;
+    }
+
+    public void setFinishTime(long finishTime) {
+        this.finishTime = finishTime;
+    }
+
+    public int getProgress() {
+        return this.progress;
+    }
+
+    public void setProgress(int progress) {
+        this.progress = progress;
+    }
+
+    /**
+     * get statisticsJob from analyzeStmt.
+     * AnalyzeStmt: analyze t1(c1,c2,c3)
+     * tableId: [t1]
+     * tableIdToColumnName <t1, [c1,c2,c3]>
+     */
+    public static StatisticsJob fromAnalyzeStmt(AnalyzeStmt analyzeStmt) throws UserException {
+        List<Long> tableIdList = Lists.newArrayList();
+        Map<Long, List<String>> tableIdToColumnName = Maps.newHashMap();
+        List<String> columnNames = analyzeStmt.getColumnNames();
+
+        String dbName = analyzeStmt.getDbName();
+        String tblName = analyzeStmt.getTblName();
+        Database db = Catalog.getCurrentCatalog().getDbOrDdlException(dbName);
+
+        if (Strings.isNullOrEmpty(tblName)) {
```

Comment:
duplicate code with AnalyzeStmt, could we save these infos in AnalyzeStmt, and reuse here?